compiler/semcall.nim

```diff
@@ -327,10 +328,30 @@ proc resolveOverloads(c: PContext, n, orig: PNode,
       pickBest(callOp)
 
     if overloadsState == csEmpty and result.state == csEmpty:
+      var msg = ""
+      if c.compilesContextId > 0:
+        # we avoid running more diagnostic when inside a `compiles(expr)`, to
+        # errors while running diagnostic (see test D20180828T234921), and
+        # also avoid slowdowns in evaluating `compiles(expr)`.
+        discard
+      else:
+        var o: TOverloadIter
+        var sym = initOverloadIter(o, c, f)
+        while sym != nil:
+          proc toHumanStr(kind: TSymKind): string=
+            result = $kind
+            if result.startsWith "sk": result = result[2..^1].toLowerAscii
+          msg &= "\n  * found '$1' of kind '$2'" % [getSymRepr(c.config, sym), sym.kind.toHumanStr]
```

Comment:
The compiler doesn't output asterisk lists.

Reply:
done